server/odc-service/src/main/java/com/oceanbase/odc/service/permission/table/TablePermissionService.java

```diff
@@ -0,0 +1,604 @@
+import com.oceanbase.odc.service.collaboration.project.ProjectService;
+import com.oceanbase.odc.service.collaboration.project.model.Project;
+import com.oceanbase.odc.service.connection.ConnectionService;
+import com.oceanbase.odc.service.connection.database.DatabaseService;
+import com.oceanbase.odc.service.connection.database.model.Database;
+import com.oceanbase.odc.service.connection.database.model.DatabaseSyncStatus;
+import com.oceanbase.odc.service.connection.model.ConnectionConfig;
+import com.oceanbase.odc.service.connection.table.TableService;
+import com.oceanbase.odc.service.connection.table.model.Table;
+import com.oceanbase.odc.service.iam.PermissionService;
+import com.oceanbase.odc.service.iam.ProjectPermissionValidator;
+import com.oceanbase.odc.service.iam.auth.AuthenticationFacade;
+import com.oceanbase.odc.service.permission.database.DatabasePermissionHelper;
+import com.oceanbase.odc.service.permission.database.model.DatabasePermissionType;
+import com.oceanbase.odc.service.permission.database.model.ExpirationStatusFilter;
+import com.oceanbase.odc.service.permission.table.model.CreateTablePermissionReq;
+import com.oceanbase.odc.service.permission.table.model.CreateTablePermissionReq.TablePermission;
+import com.oceanbase.odc.service.permission.table.model.QueryTablePermissionParams;
+import com.oceanbase.odc.service.session.model.UnauthorizedResource;
+import com.oceanbase.tools.sqlparser.statement.common.RelationFactor;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * ClassName: TablePermissionService Package: com.oceanbase.odc.service.permission.table
+ * Description:
+ *
+ * @Author: fenghao
+ * @Create 2024/3/11 20:49
+ * @Version 1.0
+ */
+@Service
+@Validated
+@Authenticated
+@Slf4j
+public class TablePermissionService {
+    @Autowired
+    private ProjectService projectService;
+
+    @Autowired
+    private ProjectPermissionValidator projectPermissionValidator;
+
+    @Autowired
+    @Lazy
+    private DatabaseService databaseService;
+
+    @Autowired
+    private AuthenticationFacade authenticationFacade;
+
+    @Autowired
+    private PermissionRepository permissionRepository;
+
+    @Autowired
+    private UserPermissionRepository userPermissionRepository;
+
+    @Autowired
+    private UserTablePermissionRepository userTablePermissionRepository;
+
+    @Autowired
+    private PermissionService permissionService;
+
+    @Autowired
+    @Lazy
+    private TableService tableService;
+
+    @Autowired
+    private ConnectionService connectionService;
+
+    @Autowired
+    private DatabasePermissionHelper databasePermissionHelper;
+
+    @Value("${odc.iam.permission.expired-retention-time-seconds:7776000}")
+    private long expiredRetentionTimeSeconds;
+
+    private static final UserTablePermissionMapper mapper = UserTablePermissionMapper.INSTANCE;
+    private static final int EXPIRING_DAYS = 7;
+
+    private static final Set<String> ORACLE_DATA_DICTIONARY = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
`ORACLE_DATA_DICTIONARY` and `MYSQL_DATA_DICTIONARY` are always empty?

Reply:
fixed

Reply:
add default value for dictionary, and modify ignoreDataDictionary  default is false.